v2/cli/event_commands.go

```diff
@@ -0,0 +1,722 @@
+					event.Type,
+					duration.ShortHumanDuration(time.Since(*event.Created)),
+					event.Worker.Status.Phase,
+				)
+			}
+			fmt.Println(table)
+
+		case flagOutputYAML:
+			yamlBytes, err := yaml.Marshal(events)
+			if err != nil {
+				return errors.Wrap(
+					err,
+					"error formatting output from get events operation",
+				)
+			}
+			fmt.Println(string(yamlBytes))
+
+		case flagOutputJSON:
+			prettyJSON, err := json.MarshalIndent(events, "", "  ")
+			if err != nil {
+				return errors.Wrap(
+					err,
+					"error formatting output from get events operation",
+				)
+			}
+			fmt.Println(string(prettyJSON))
+		}
+
+		if events.RemainingItemCount < 1 || events.Continue == "" {
+			break
+		}
+
+		// Exit after one page of output if this isn't a terminal
+		if !terminal.IsTerminal(int(os.Stdout.Fd())) {
+			break
+		}
+
+		if shouldContinue, err :=
+			shouldContinue(events.RemainingItemCount); err != nil {
+			return err
+		} else if !shouldContinue {
+			break
+		}
+
+		opts.Continue = events.Continue
+	}
+
+	return nil
+}
+
+func eventGet(c *cli.Context) error {
+	id := c.String(flagID)
+	output := c.String(flagOutput)
+
+	if err := validateOutputFormat(output); err != nil {
+		return err
+	}
+
+	client, err := getClient(c)
+	if err != nil {
+		return err
+	}
+
+	event, err := client.Core().Events().Get(c.Context, id)
+	if err != nil {
+		return err
+	}
+
+	switch strings.ToLower(output) {
+	case flagOutputTable:
+		table := uitable.New()
+		table.AddRow("ID", "PROJECT", "SOURCE", "TYPE", "AGE", "WORKER PHASE")
+		var age string
+		if event.Created != nil {
+			age = duration.ShortHumanDuration(time.Since(*event.Created))
+		}
+		table.AddRow(
+			event.ID,
+			event.ProjectID,
+			event.Source,
+			event.Type,
+			age,
+			event.Worker.Status.Phase,
+		)
+		fmt.Println(table)
+
+		if len(event.Worker.Jobs) > 0 {
+			fmt.Printf("\nEvent %q jobs:\n\n", event.ID)
+			table = uitable.New()
+			table.AddRow("NAME", "STARTED", "ENDED", "PHASE")
+			for jobName, job := range event.Worker.Jobs {
+				jobStatus := job.Status
+				var started, ended string
+				if jobStatus.Started != nil {
+					started =
+						duration.ShortHumanDuration(time.Since(*jobStatus.Started))
+				}
+				if jobStatus.Ended != nil {
+					ended =
+						duration.ShortHumanDuration(time.Since(*jobStatus.Ended))
+				}
+				table.AddRow(
+					jobName,
+					started,
+					ended,
+					jobStatus.Phase,
+				)
+			}
+			fmt.Println(table)
+		}
+
+	case flagOutputYAML:
+		yamlBytes, err := yaml.Marshal(event)
+		if err != nil {
+			return errors.Wrap(
+				err,
+				"error formatting output from get event operation",
+			)
+		}
+		fmt.Println(string(yamlBytes))
+
+	case flagOutputJSON:
+		prettyJSON, err := json.MarshalIndent(event, "", "  ")
+		if err != nil {
+			return errors.Wrap(
+				err,
+				"error formatting output from get event operation",
+			)
+		}
+		fmt.Println(string(prettyJSON))
+	}
+
+	return nil
+}
+
+func eventCancel(c *cli.Context) error {
```

Comment:
Thought: Do we want a flag around whether or not to delete workers and jobs for `Cancel` and `CancelMany`?  In reviewing the implementations, it appears `DeleteWorkerAndJobs` is called by default... are there use cases where users might want to cancel an event but not delete these?

Reply:
> Do we want a flag around whether or not to delete workers and jobs for Cancel and CancelMany?

I don't think so, because...

> it appears `DeleteWorkerAndJobs` is called by default...

Right... This deletes running stuff (if there is any) _from the substrate_. i.e. You cancel an Event and the worker, pod, etc. don't go away from the db, but whatever (if anything) related to it is running on the substrate, it should be removed from there.

Reply:
👍 Ok, thank you for the clarification.